lib/src/multi_search_query.dart

```diff
@@ -0,0 +1,13 @@
+import 'search_query.dart';
+
+class MultiSearchQuery {
+  final List<SearchQuery> queries;
+
+  const MultiSearchQuery({
+    required this.queries,
+  });
+
+  Map<String, Object?> toMap() {

```

Comment:
Does this work?

```suggestion
  Map<String, SearchQuery?> toMap() {
```

Reply:
nope, it should be as is to be backward compatible with any future changes .
also it would be `Map<String, List<SearchQuery>>`